packages/shared/src/archive-storybook/index.ts

```diff
@@ -28,8 +28,13 @@ export function buildArchiveStorybook(
 }
 
 function binPath() {
-  // eslint-disable-next-line global-require
+  // eslint-disable-next-line @typescript-eslint/no-require-imports
   const packageJson = require('storybook/package.json');
+  const parsedVersion = parseInt(packageJson.version[0], 10);
+  const storybookPackageJsonHasBinObject = parsedVersion >= 9;
   // reference the entry file based on the package.json `bin` value, since it changed between SB 8.1 and 8.2
-  return resolve(dirname(require.resolve('storybook/package.json')), packageJson.bin.storybook);
+  return resolve(
```

Comment:
You might want to try a similar approach to this:
https://github.com/storybookjs/test-runner/blob/next/src/test-storybook.ts#L128

So that the resolution will start from the node_modules of the e2e package instead. If a user had something like this:
```bash
/node_modules
|_/storybook
|_/chromatic-e2e
|__/node_modules
|___/storybook
```

it will prefer the one from chromatic-e2e

Reply:
@skitterm @tevanoff have you considered my suggestion?

Reply:
@yannbf we decided to forego putting e2e on Storybook 9 for the time being. I'll close this PR to reflect that.